src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScript.java

```diff
@@ -159,23 +307,81 @@ public Object evaluate(ClassLoader loader, Binding binding) throws Exception {
         }
         try {
             loader = GroovySandbox.createSecureClassLoader(loader);
+
+            Field loaderF = GroovyShell.class.getDeclaredField("loader");
```

Comment:
You need to catch `NoSuchFieldException` and quietly fall back gracefully, in case this code is refactored in a future version of Groovy. (I do _not_ recommend swallowing other reflection exceptions—log them as `WARNING`s: they are not expected. I.e., do not repeat this mistake: https://github.com/jenkinsci/cloudbees-folder-plugin/pull/79#discussion_r144733904)